### ISSUE TITLE
[1.10.2] Dupe bug with barrel

### ISSUE THREAD

**Issue by Blatts12** [maruohon/enderutilities#56]
1. Have less than stack of item on hotbar (for me it's apple).
2. Take stack from barrel.
3. Have 2 stack of item.

I don't test that only with EnderUtilities.

**Comment by maruohon** [comment]
How often does this happen for you?

I did see similar behavior in my own testing a while back, but after several hours of tedious debugging, I narrowed it down to some weird vanilla issue, where sometimes the player picks up the same item from the ground more than once. It seems to happen when either the item that is being picked up has lots of NBT data on it, or when the player has items in their inventory that have lots of NBT data. I have no idea why that makes a difference, unless it's some timing related thing, and that large NBT data simply makes the inventory compare & sync code take longer and then the bug appears.

So basically there is nothing I can do about it, and it's actually on no way related to the barrel. It should also happen when you simply pick up any item from the ground. But it can be rare, and like I mentioned, it starts to happen more and more when you have items with lots of NBT data on them, like bags filled with items.
